### ISSUE TITLE
Hourly time is offset by 30 minutes in certain locations

### ISSUE THREAD

**Issue by guillaume** [Pirate-Weather/pirateweather#32]
```
# mumbai Asia/Kolkata timezone does not return hourly time for requested time 
# offset is 5.5
% curl -s "https://api.pirateweather.net/forecast/<REDACTED>/19.0785451,72.878176,1678732200?&units=si&extend=hourly&tz=precise" | jq '.hourly.data[0].time'
1678730400

# Usually that will return the time requested
# offset is -4 for "America/Toronto"
% curl -s "https://api.pirateweather.net/forecast/<REDACTED>/40.7127281,-74.0060152,1678680000?&units=si&extend=hourly&tz=precise" | jq '.hourly.data[0].time'
1678680000
```

**Comment by cloneofghosts** [comment]
Looks like this issue was missed but I tested and I can confirm that for fractional timezones the `time` parameter is not aligned with the top of the hour but the daily block works fine.

A city to test this with is St. John's, Newfoundland (47.486771,-52.751787) but you have to use tz=precise to demonstrate the issue otherwise the timezone is completely incorrect.

```
  "latitude": 47.486771,
  "longitude": -52.751787,
  "timezone": "Etc/GMT+4",
  "offset": -4.0,
```
VS
```
  "latitude": 47.486771,
  "longitude": -52.751787,
  "timezone": "America/St_Johns",
  "offset": -2.5,
```

I'll tag @alexander0042 so he can look into this.

**Comment by cloneofghosts** [comment]
Will bump this one last time. @alexander0042 maybe this can be sorted for the 2.0 release?

**Comment by alexander0042** [comment]
Yup! Thanks for the bump. I've tagged it to keep it alive, 

Just to clarify here- is the issue that these locations always return the incorrect hourly times, or only when specific times are requested?

**Comment by cloneofghosts** [comment]
I believe this was already fixed in the TimeMachine but when I request data for St. John's, Newfroundland all the times in the hourly block are off by 30m.

I did a test query and the first block in the hourly section shows `"time": 1697144400,` and converting that into a proper timestamp you get:

1697144400 converts to Thursday October 12, 2023 18:30:00 (pm) in time zone America/St Johns (NDT)

**Comment by alexander0042** [comment]
The classic Gander issue! Fixed this in 2.0i along with my other timezone bugs

**Comment by cloneofghosts** [comment]
Did a quick test and can confirm that this is fixed so will close this issue.
